src/dbnode/digest/writer.go

```diff
@@ -109,3 +112,37 @@ func (w *fdWithDigestContentsWriter) WriteDigests(digests ...uint32) error {
 	}
 	return nil
 }
+
+func CloseAll(writers ...FdWithDigestWriter) error {
+	multiErr := xerrors.NewMultiError()
+	for _, writer := range writers {
+		if err := writer.Close(); err != nil {
+			multiErr = multiErr.Add(err)
+		}
+	}
+	return multiErr.FinalError()
+}
+
+func SyncAll(writers ...FdWithDigestWriter) error {
+	var (
+		multiErr = xerrors.NewMultiError()
+		errMut   = sync.Mutex{}
+		wg       sync.WaitGroup
+	)
+
+	wg.Add(len(writers))
+	for _, writer := range writers {
```

Comment:
nit: may be slightly safer for sake of future compat to do single wg updates per loop
```suggestion
	for _, writer := range writers {
		wg.Add(1)
```

Reply:
👍